test/suites/integration/deregister_targets_test.go

```diff
@@ -83,16 +83,16 @@ var _ = Describe("Deregister Targets", func() {
 		Eventually(func(g Gomega) {
 			log.Println("Verifying VPC lattice Targets deregistered")
 			targets := testFramework.GetTargets(ctx, targetGroup, deployment)
-			Expect(len(targets) == 0)
-		}).WithTimeout(5*time.Minute + 10*time.Second)
+			g.Expect(targets).To(HaveLen(0))
+		}).WithTimeout(5*time.Minute + 10*time.Second).Should(Succeed())
```

Comment:
A general comments about eventually. Please use eventually as assertion block and not side effect of it - a nested assertions.
```
Eventually(func() int {
  targets, err := latticeClient.getTargets()
  if err != nil { return -1 }
  return len(targets)
}).WithPolling(time.Minute).WithTimeout(10*time.Minutes).Should(Equal(0))
```